docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/MulticlassClassification/LbfgsMaximumEntropy.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML;
+using Microsoft.ML.Data;
+
+namespace Samples.Dynamic.Trainers.MulticlassClassification
+{
+    public static class LbfgsMaximumEntropy
+    {
+        public static void Example()
+        {
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+            // as a catalog of available operations and as the source of randomness.
+            // Setting the seed to a fixed number in this example to make outputs deterministic.
+            var mlContext = new MLContext(seed: 0);
+
+            // Create a list of training data points.
+            var dataPoints = GenerateRandomDataPoints(1000);
+
+            // Convert the list of data points to an IDataView object, which is consumable by ML.NET API.
+            var trainingData = mlContext.Data.LoadFromEnumerable(dataPoints);
+
+            // Define the trainer.
+            var pipeline =
+                    // Convert the string labels into key types.
+                    mlContext.Transforms.Conversion.MapValueToKey("Label")
```

Comment:
>MapValueToKey [](start = 52, length = 13)

I'm not sure if this is an issue, there may be other samples for this, but would it make sense to pass the `IDataView keyData` argument to this method to show how the user can avoid a pass over the data to get the labels in case the set of labels is known?

Reply:
I think it's important to show that use of the method, and I hope that there is a sample for that under `MapValueToKey`. However, I don't think it would be the right place here to include such a sample.
What I could do instead is load a keyType directly, instead of using MapValueToKey. Would that be better?

---
In reply to: [275474158](https://github.com/dotnet/machinelearning/pull/3322#discussion_r275474158) [](ancestors = 275474158)